model/src/main/java/com/flipkart/flux/domain/Context.java

```diff
@@ -52,15 +54,45 @@
      */
     public abstract Object retrieve(String key);
 
-    public List<State<T>> getExecutableStates(State<T> currentState, Event<T> event) {
-        // Go through the dependency graph to figure the states that can now be executed
-        return null;
+    public Set<State> getDependantStates(String eventName) {
+        return stateToEventDependencyGraph.get(eventName);
     }
+
     public boolean isExecutionCancelled() {
         //check for cancelledException in data, and return whether state machine execution is cancelled or not
         return false;
     }
 
+    /**
+     * This builds dependency graph between event and states and keeps for later use. Currently dependency graph is created on every event arrival.
+     */
+    public void buildDependencyMap(Set<State> states) {
+        stateToEventDependencyGraph = new HashMap<>();
+        for(State state : states) {
+            if (state.getDependencies() != null) {
```

Comment:
@yogeshnachnani I think it would be better if we keep event name as "start" if the state is not depending on any event.  In dependency graph also we can use the same.